### PR TITLE
Add support for Adam(W) and weight decay 

### PR DESCRIPTION
**Pull request by jveitchmichaelis**
## Description

Adds support for different optimizers in the config. Shouldn't break anything backwards as this was never exposed in older versions of DeepForest.

Fixes a small bug/footgun in the scheduler config where if you specified something not in the list, it would silently default to  ReduceLROnPlateau.

Adam/w is widely used for training transformer-based models, and also supports a weight_decay parameter. 

## Related Issue(s)

In support of newer models being added to DeepForest.

## AI-Assisted Development

- [X] I used AI tools (e.g., GitHub Copilot, ChatGPT, etc.) in developing this PR
- [X] I understand all the code I'm submitting
- [X] I have reviewed and validated all AI-generated code

**AI tools used (if applicable):**

Claude

### CODE REVIEW

**Comment by bw4sz** [comment]
Yes, I've meant to do this. I sometimes wonder if there isn't a better kwargs type way of getting at pytorch-lightning, I don't love this pattern where we just wrap all things into the config. But I guess it does meet the style of trying to set reasonable defaults for users?

**Review by bw4sz — APPROVED** [review]
I'm approving this and we can discuss the larger pattern at Thursday meeting.

**Comment by jveitchmichaelis** [comment]
Anything that's a trainer arg specifically can probably be a config group e.g. `+trainer.strategy= ... ` and then we get CLI support for limit_batches and other things.

(Ie none of these options would be in the config, you'd pass a dict that gets added onto whatever is passed to the trainer, sort of like a **kwargs)
